### PR TITLE
Support base field using host from CLI

### LINKED ISSUE #23: Need `base` member or absolute URLs in `href` members
When I try to add the lamp Thing to my WoT Consumer, it complains that the Thing Description is not valid because the values of `href` are not valid URLs.

The reason for this is that the values of `href` are paths, not full URLs, and there's no `base` member with a root URL to resolve them against.

I assume that you intend for the paths to be relative to the root of the domain, but once a Thing Description is added to a directory there is no way for a Consumer to know what the source origin was. It's therefore not possible for a Consumer to fetch any of the URLs provided in Forms.

I think you need to either:
- Add a `base` member which provides the root URL against which `href` members should be resolved
- Make `href` members absolute URLs

Note that this requires that the server knows the host from which the Thing Description is being served, which will not always be localhost.

### PR DESCRIPTION
**Pull request by dodomorandi**
This should fix #23. @benfrancis sorry for bothering, could you check whether this is sufficient to solve the issue? I added a `--host` CLI param which is then used to fill the `base` field in the WoT description.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -67,6 +74,30 @@ impl CliCommon {
     pub fn socket_addr(&self) -> SocketAddr {
         SocketAddr::from((self.bind_addr, self.listen_port))
     }
+
+    pub fn set_thing_base<O: ExtendableThing, S>(
+        &self,
+        thing_builder: ThingBuilder<O, S>,
+    ) -> ThingBuilder<O, S> {
+        match &self.host {
+            Some(host) => thing_builder.base(format!("http://{host}/")),
```
Comment (lu-zero) [inline src/lib.rs:83]: It needs the port as well, I think

Comment (dodomorandi) [reply src/lib.rs:83]: Good point!

**Review by lu-zero — APPROVED** [review]
It needs a rebase, I'd go with it for now.

**Comment by benfrancis** [comment]
Sorry for the late response.

I can confirm that this works (once I figured out how to use it), and I can now consume the lamp web thing with my Consumer :) Sneak preview:

Screen Shot 2023-04-14 at 11 15 48


For anyone trying to use this, this is the command that worked:

`$ cargo run --bin lamp -- --host localhost`

A TD can then be found at http://localhost:3000/.well-known/wot with a `base` value of `http://localhost:3000/`

**Pull request merged**